subprojects/core/src/integTest/groovy/org/gradle/api/services/BuildServiceIntegrationTest.groovy

```diff
@@ -56,17 +56,29 @@ class BuildServiceIntegrationTest extends AbstractIntegrationSpec {
             }
         """
 
+        if (enableFeaturePreview) {
+            settingsFile '''
+                enableFeaturePreview 'STABLE_BUILD_SERVICES'
+            '''
+            executer.expectDocumentedDeprecationWarning(
+                "Build service 'counter' is being used by task ':broken' without the corresponding declaration via 'Task#usesService'. " +
+                    "This will fail with an error in Gradle 8.0. " +
+                    "Max parallel usages constraint on the build service can't be honored. " +
```

Comment:
I think we should add an entry to the upgrade guide for this deprecation and move this kind of detail there, where we can provide more context. We can do this in a later PR, though.

Reply:
👍